transcom/mymove
# MB- 15179-Reason-required-For-All- SIT

## [Jira ticket](https://dp3.atlassian.net/browse/MB-15947)

## Summary

The purpose of the ticket is to ensure that reason is required for all SIT service items (destination and origin sit). In the API we have made reason required in the origin site, but not in destination SIT. `Prime.yaml` has been updated to reflect this correction, and validation has been added in the `payload_to_model` for both origin and destination SIT. The `reason` field has also been added where it was previously missing in other files.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### How to test

1. Check that the prime.yaml, Payload_to_model, and model_to_payload have been updated for SIT service items
2. Run the test in tests:
```
go test ./pkg/handlers/primeapi/payloads --testify.m TestMTOServiceItemModel -v
```

```
go test ./pkg/handlers/primeapi--testify.m TestCreateMTOServiceItemHandler -v
```

```
go test ./pkg/handlers/primeapi--testify.m TestCreateMTOServiceItemDestSITHandler -v
```

### Backend
